Bottom: 54c0581f679958b4defd45805ec45fac3f08281c
Top:    c5d8e80458728819cdc59826bc3afd61a953d775
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-25 01:41:48 +0100

hid/gtk: Fix up paning when flipping the board.

I'm not sure when this got broken, but it was probably my fault ;)


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index c917e43..83df23f 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -1458,12 +1458,12 @@ SwapSides (int argc, char **argv, int x, int y)
   if (do_flip_x)
     {
 	flipd = PCB->MaxWidth / 2 - SIDE_X (gport->pcb_x);
-	ghid_port_ranges_pan (2 * flipd, 0, TRUE);
+	ghid_port_ranges_pan (-2 * flipd, 0, TRUE);
     }
   if (do_flip_y)
     {
 	flipd = PCB->MaxHeight / 2 - SIDE_Y (gport->pcb_y);
-	ghid_port_ranges_pan (0, 2 * flipd, TRUE);
+	ghid_port_ranges_pan (0, -2 * flipd, TRUE);
     }
 
   ghid_invalidate_all ();
